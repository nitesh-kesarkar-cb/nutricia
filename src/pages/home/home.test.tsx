import { render, screen } from '@testing-library/react';
import HomePage from './home';

describe('HomePage', () => {
   it('renders the main heading', () => {
      render(<HomePage />);
      // h2 with partial, case-insensitive match
      const heading = screen.getByRole('heading', {
         level: 2,
         name: /react \+ vite \+ tailwind/i,
      });
      expect(heading).toBeInTheDocument();
   });

   it('renders the lead paragraph', () => {
      render(<HomePage />);
      expect(
         screen.getByText(/production-ready boilerplate/i)
      ).toBeInTheDocument();
   });

   it('renders all five buttons with correct labels', () => {
      render(<HomePage />);

      // Each button by accessible name
      expect(
         screen.getByRole('button', { name: /primary/i })
      ).toBeInTheDocument();
      expect(
         screen.getByRole('button', { name: /secondary/i })
      ).toBeInTheDocument();
      expect(
         screen.getByRole('button', { name: /outline/i })
      ).toBeInTheDocument();
      expect(
         screen.getByRole('button', { name: /ghost/i })
      ).toBeInTheDocument();
      expect(screen.getByRole('button', { name: /link/i })).toBeInTheDocument();

      // And exactly 5 buttons total
      expect(screen.getAllByRole('button')).toHaveLength(5);
   });

   it('matches snapshot', () => {
      const { asFragment } = render(<HomePage />);
      expect(asFragment()).toMatchSnapshot();
   });
});
