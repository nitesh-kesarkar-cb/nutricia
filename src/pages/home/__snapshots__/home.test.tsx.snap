// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage > matches snapshot (keys, not translations) 1`] = `
<DocumentFragment>
  <section
    class="space-y-6"
  >
    <div>
      <h2
        class="text-3xl font-bold tracking-tight"
      >
        home.title
      </h2>
      <p
        class="mt-2 text-muted-foreground"
      >
        home.lead
      </p>
    </div>
    <div
      class="flex gap-3"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 ring-offset-background bg-primary text-primary-foreground hover:opacity-90 h-10 px-4 py-2"
      >
        home.buttons.primary
      </button>
    </div>
  </section>
</DocumentFragment>
`;

exports[`HomePage > matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="space-y-6"
  >
    <div>
      <h2
        class="text-3xl font-bold tracking-tight"
      >
        home.title
      </h2>
      <p
        class="mt-2 text-muted-foreground"
      >
        home.lead
      </p>
    </div>
    <div
      class="flex gap-3"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 ring-offset-background bg-primary text-primary-foreground hover:opacity-90 h-10 px-4 py-2"
      >
        home.buttons.primary
      </button>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 ring-offset-background bg-secondary text-secondary-foreground hover:opacity-90 h-10 px-4 py-2"
      >
        home.buttons.secondary
      </button>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 ring-offset-background border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
      >
        home.buttons.outline
      </button>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 ring-offset-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
      >
        home.buttons.ghost
      </button>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 ring-offset-background text-primary underline-offset-4 hover:underline h-10 px-4 py-2"
      >
        home.buttons.link
      </button>
    </div>
  </section>
</DocumentFragment>
`;
